### ISSUE TITLE
[enhancement] preserve the real status code

### ISSUE THREAD

**Issue by GeoffWilliams** [bradfeehan/derelict#16]
Is derelict still an active project?

If so, it would be great for the library to provide a way for users to obtain the real status code after running a command.  This is really useful when combined with `vagrant ssh -c` because it means I can run commands on the guest VM and get the real status code back which I need for another library I'm developing

**Comment by bradfeehan** [comment]
If you use `Instance.execute!` (which most internal parts of the library do), you should get an exception if the command failed.

See: https://github.com/bradfeehan/derelict/blob/master/lib/derelict/instance.rb#L78-L85

**Comment by GeoffWilliams** [comment]
I had a look at the code linked - the problem I'm trying to solve is that I need the numerical exit status of the command I'm running, not just whether it passed or failed.  The unless I'm missing something, the executor only saves a true/false value, I added the status code in the PR attached to this issue.

Just to give some background on this, I'm running puppet in `--detailed-exitcodes` mode where the following exit codes may be returned:
* 0: The run succeeded with no changes or failures; the system was already in the desired state.
* 1: The run failed, or wasn't attempted due to another run already in progress.
* 2: The run succeeded, and some resources were changed.
* 4: The run succeeded, and some resources failed.
* 6: The run succeeded, and included both changes and failures.

Long story short I need to run the same command twice:
* 1st time it runs, exit status 0 or 2 are acceptable
* 2nd time it runs, exit status must be 0

So I figured exposing the status code was the way to go.  Does that make more sense?  Is there another way to get the exit code (numerical) from the Executor instance in the exception that I'm missing?

- thanks!

**Comment by bradfeehan** [comment]
Ah, I see the difference -- I think you're on the right track.

BTW I should warn you, this project is quite old and may not work with latest Vagrant versions, so your efforts may be in vain (unless you already know this library works for what you want).

**Comment by GeoffWilliams** [comment]
Yeah I did wonder that when I saw the name.  Surprisingly it works like a champ and everything else in the library does exactly what I want.  I'm using a patched version of the library (this PR) and have had no problems at all - latest vagrant on latest ubuntu.  I've temporarily published a fix on rubygems.org just so that I could depend on a fixed version but it would be great if this gets merged as I'd be able to delete it
